src/EventListener/ReadListener.php

```diff
@@ -56,10 +59,12 @@ public function onKernelRequest(GetResponseEvent $event)
             return;
         }
 
-        if (isset($attributes['collection_operation_name'])) {
-            $data = $this->getCollectionData($request, $attributes);
-        } else {
-            $data = $this->getItemData($request, $attributes);
+        foreach (OperationTypes::TYPES as $operationType) {
+            if (isset($attributes["{$operationType}_operation_name"])) {
+                $method = sprintf('get%sData', ucfirst($operationType));
```

Comment:
If would prefer a switch or something like that instead of dynamic call (it's hard to debug and it prevents PHP to optimize at compile time).